pkg/controllers/controllers.go

```diff
@@ -78,9 +81,11 @@ func NewControllers(
 ) []controller.Controller {
 	p := provisioning.NewProvisioner(kubeClient, recorder, cloudProvider, cluster, clock)
 	evictionQueue := terminator.NewQueue(kubeClient, recorder)
-	disruptionQueue := disruption.NewQueue(kubeClient, recorder, cluster, clock, p)
 	npState := nodepoolhealth.NewState()
 	clusterCost := cost.NewClusterCost(ctx, cloudProvider, kubeClient)
+	podResources := podresources.NewPodResources()
+	tracker := disruption.NewTracker(cluster, clusterCost, podResources, clock, cache.New(30*time.Minute, time.Minute), nil, false)
+	disruptionQueue := disruption.NewQueue(kubeClient, recorder, cluster, clock, p, tracker)
```

Comment:
It doesn't look like we've actually added a mechanism to allow cloud providers to enable this, have we? Seems like a good use for a functional opt here:
```go
NewControllers(..., EnableDisruptionTracking())
```